File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/repository/custom/UserCustomExtendedRepository.java
package com.coupang.springframework.data.requery.repository.custom;

import com.coupang.springframework.data.requery.domain.basic.BasicUser;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

public interface UserCustomExtendedRepository extends CustomGenericRepository<BasicUser, Long> {

    @Transactional(readOnly = false, timeout = 10)
    List<BasicUser> findAll();


    @Transactional(readOnly = false, timeout = 10)
    Optional<BasicUser> findById(Long id);

}

File: spring-data-requery-kotlin/src/test/kotlin/org/springframework/data/requery/kotlin/repository/custom/UserCustomExtendedRepository.kt
package org.springframework.data.requery.kotlin.repository.custom

import org.springframework.data.requery.kotlin.domain.sample.User
import org.springframework.transaction.annotation.Transactional
import java.util.*

interface UserCustomExtendedRepository: CustomGenericRepository<User, Int> {

    @Transactional(readOnly = false, timeout = 10)
    override fun findAll(): List<User>

    @Transactional(readOnly = false, timeout = 10)
    override fun findById(id: Int): Optional<User>
}